plugin/trino-http-query-listener/src/main/java/io/trino/plugin/httpquery/HttpQueryListenerConfig.java

```diff
@@ -0,0 +1,206 @@
+    }
+
+    @ConfigDescription("Will log io.trino.spi.eventlistener.QueryCompletedEvent")
+    @Config("http-query-listener.log-completed")
+    public HttpQueryListenerConfig setLogCompleted(boolean logCompleted)
+    {
+        if (logCompleted) {
+            loggedEvents.add(HttpQueryListenerEventType.COMPLETED);
+        }
+        return this;
+    }
+
+    public boolean getLogCompleted()
+    {
+        return loggedEvents.contains(HttpQueryListenerEventType.COMPLETED);
+    }
+
+    @ConfigDescription("Will log io.trino.spi.eventlistener.SplitCompletedEvent")
+    @Config("http-query-listener.log-split")
+    public HttpQueryListenerConfig setLogSplit(boolean logSplit)
+    {
+        if (logSplit) {
+            loggedEvents.add(HttpQueryListenerEventType.SPLIT);
+        }
+        return this;
+    }
+
+    public boolean getLogSplit()
+    {
+        return loggedEvents.contains(HttpQueryListenerEventType.SPLIT);
+    }
+
+    public URL getIngestUrl()
+    {
+        return ingestUrl;
+    }
+
+    @ConfigDescription("URL of receiving server. Explicitly set the scheme https:// to use symmetric encryption")
+    @Config("http-query-listener.connect-ingest-url")
+    public HttpQueryListenerConfig setIngestUrl(String ingestUrl)
+            throws MalformedURLException
+    {
+        this.ingestUrl = new URL(ingestUrl);
+        return this;
+    }
+
+    @FileExists
+    public String getSslCertStorePath()
+    {
+        return sslCertStorePath;
+    }
+
+    @ConfigDescription("Path to PKCS 12 certificate store")
+    @Config("http-query-listener.connect-ssl-cert-store-path")
+    public HttpQueryListenerConfig setSslCertStorePath(String sslCertStorePath)
+    {
+        this.sslCertStorePath = sslCertStorePath;
+        return this;
+    }
+
+    public String getSslCertStorePassword()
+    {
+        return sslCertStorePassword;
+    }
+
+    @ConfigDescription("Password of PKCS 12 certificate store")
+    @Config("http-query-listener.connect-ssl-cert-store-password")
+    public HttpQueryListenerConfig setSslCertStorePassword(String sslCertStorePassword)
+    {
+        this.sslCertStorePassword = sslCertStorePassword;
+        return this;
+    }
+
+    public Map<String, String> getHttpHeaders()
+    {
+        return httpHeaders;
+    }
+
+    @ConfigDescription("List of custom custom HTTP headers provided as: \"Header-Name-1: header value 1, Header-Value-2: header value 2, ...\" ")
+    @Config("http-query-listener.connect-http-headers")
+    public HttpQueryListenerConfig setHttpHeaders(List<String> httpHeaders)
+    {
+        try {
+            this.httpHeaders = httpHeaders
+                    .stream()
+                    .collect(Collectors.toMap(kvs -> kvs.split(":", 2)[0], kvs -> kvs.split(":", 2)[1]));
+        }
+        catch (IndexOutOfBoundsException e) {
+            throw new IllegalArgumentException(String.format("Cannot parse http headers from %s", String.join(", ", httpHeaders)), e);
```

Comment:
Please verify if invalid format here results in user-readable error which includes property name.